agent/worker.go

```diff
@@ -278,8 +278,14 @@ func reconcileTaskState(ctx context.Context, w *worker, assignments []*api.Assig
 
 	removeTaskAssignment := func(taskID string) error {
 		ctx := log.WithLogger(ctx, log.G(ctx).WithField("task.id", taskID))
-		if err := SetTaskAssignment(tx, taskID, false); err != nil {
-			log.G(ctx).WithError(err).Error("error setting task assignment in database")
+		// if a task is no longer assigned, then we do not have to keep track
+		// of it. a task will only be unassigned when it is deleted on the
+		// manager. insteaad of SetTaskAssginment to true, we'll just remove
+		// the task now.
+		if err := DeleteTask(tx, taskID); err != nil {
+			log.G(ctx).WithError(err).Errorf(
```

Comment:
Are there any situations known where deleting fails? Should entries be unassigned before deleting, so that on failure to delete, they could be cleaned up after?

Reply:
I think the only way they fail to be deleted is if they don't exist.